tests/base_tests.py

```diff
@@ -114,6 +114,30 @@ def test_columns_non_string_or_list_error(
             uninitialized_transformers[self.transformer_name](**args)
 
 
+class WeightColumnInitTests(GenericInitTests):
```

Comment:
Yup Makes sense! I've added WeightColumnFitTests class to base_tests.py in commit [69cbb07](https://github.com/lvgig/tubular/pull/239/commits/69cbb0745a985647cf13bb99e13d429d13b968e4).
Regarding generalisable tests involving weights for transform, are 'test_bad_values_in_weights_error' and 'test_weight_not_in_X_error' tests needed for transform since these are already captured in WeightColumnFitTests class?

Reply:
nice thanks!

Yeah good point re transform, think you're right and it's unnecessary as weights aren't generally used at that stage